register.lua

```diff
@@ -200,9 +192,17 @@ function courseplay:foldableLoad(savegame)
 
 	self.cp.foldingPartsStartMoveDirection = Utils.getNoNil(startMoveDir, 0);
 end;
+-- overwrite/append functions of basegame
 Foldable.load = Utils.appendedFunction(Foldable.load, courseplay.foldableLoad);
+-- NOTE: using loadFinished() instead of load() so any other mod that overwrites Vehicle.load() doesn't interfere
+Vehicle.loadFinished = Utils.overwrittenFunction(Vehicle.loadFinished, courseplay.vehiclePostLoadFinished);
```

Comment:
I think the idea was to keep the overwrite/append/prepend together with the function we are overwriting with. Not sure why you moved these here.

Reply:
I understand the idea behind that.

My thoughts are that we keep every kind of operation and so on in one place. So all the functions in one section, all variables in one section and the overwrites in one section. 
This way you have an easier time looking for something of a specific type. And if you collect all overwriting-operations in one place you just have to look and see directly which functions are overwritten instead of have to scroll through the whole file.

Reply:
This is really a question of personal preference. 

The real problem here is that you have three different changes (a bug fix, the refill enhancement plus everything from #7445) all in a this pull request. I'd prefer having these as separate pull requests and rebasing your branch after they are merged.

Reply:
of course it's better to have the different changes in different PRs. I just made the mistake that i did the formatting changes on the master-branch and so every commit came to the new branch as well where I implemented the fix that is the purpose of this PR. 
But to fix that we could just jerry-pick only the changes in `FillableFieldWorkAIDriver.lua`.

Reply:
i changed it to the way that only the fix and the addition in the gitignore are inside the PR